apps/backend/src/schemas/system_metrics.py

```
@@ -15,57 +18,57 @@ class SystemMetricBase(BaseModel):
     # Additional metrics
-    additional_metrics: Dict[str, Any] = Field(
+    additional_metrics: dict[str, Any] = Field(
         default_factory=dict, description="Additional metric data"
     )
 
     @field_validator("memory_available_bytes")
     @classmethod
-    def validate_memory_available(cls, v, info):
+    def validate_memory_available(cls, v: int | None, info: Any) -> int | None:
         if v is not None and "memory_total_bytes" in info.data and info.data["memory_total_bytes"]:
             if v > info.data["memory_total_bytes"]:
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)